proposals/lip-0045.md

```diff
@@ -1384,187 +1388,100 @@ def isChainIDAvailable(chainID: ChainID) -> bool:
-  * All `blsKey` properties must have length `BLS_PUBLIC_KEY_LENGTH` and must be pairwise distinct.  
-  * The `bftWeight` property of each element must be positive integer.
-  * Let `totalWeight` be the sum of the `bftWeight` property of every element in `activeValidators`. Then `totalWeight` has to be less than or equal to `MAX_UINT64`.
-  * Check that `totalWeight//3 + 1 <= certificateThreshold <= totalWeight`, where `//` indicates integer division.
-  * Check that the corresponding `validatorsHash` stored in `chainAccount.lastCertificate.validatorsHash` matches with the value computed from `activeValidators` and `certificateThreshold`.
-* On a sidechain, if a chain account for another sidechain is present (`storeKey != getMainchainID()`), then a chain account for the mainchain (`storeKey == getMainchainID()`) must be present, as well as the own chain account.
-* For each entry `chainData` in `interoperabilityAsset.chainDataSubstore`:
-    * Check that `chainData.storeKey[0] == ownChainAccount.chainID[0]` and `chainData.storeKey != ownChainAccount.chainID`.
-    * Check that `chainData.lastCertificate.timestamp < g.header.timestamp.
-* For each entry `channelData` in `interoperabilityAsset.channelDataSubstore`, let `channel = channelData.storeValue`:
-    * Either `channel.messageFeeTokenID==Token.getTokenIDLSK()` (corresponding to the LSK token), or `Token.getChainID(channel.messageFeeTokenID)` must be equal to `channelData.storeKey` or `ownChainAccount.chainID` (the message fee token must be a native token of either chains).
-* On a sidechain, `interoperabilityAsset.registeredNamesSubstore` must be empty.
-* For every substore, create all the corresponding entries in the Interoperability module state.
+#### Genesis Assets Schema
 
 ```java
 genesisInteroperabilityStoreSchema = {
     "type": "object",
     "required": [
-        "outboxRootSubstore",
-        "chainDataSubstore",
-        "channelDataSubstore",
-        "chainValidatorsSubstore",
-        "ownChainDataSubstore",
-        "terminatedStateSubstore",
-        "terminatedOutboxSubstore",
-        "registeredNamesSubstore"
+        "ownChainName",
+        "ownChainNonce",
+        "chainInfos",
+        "terminatedStateAccounts",
+        "terminatedOutboxAccounts"
     ],
     "properties": {
-        "outboxRootSubstore": {
-            "type": "array",
-            "fieldNumber": 1,
-            "items": {
-                "type": "object",
-                "required": ["storeKey", "storeValue"],
-                "properties": {
-                    "storeKey": {
-                        "dataType": "bytes",
-                        "fieldNumber": 1
-                    },
-                    "storeValue": {
-                        ...outboxRootSchema,
-                        "fieldNumber": 2,
-                    }
-                }
-            }
+        "ownChainName": {
+            "dataType": "string",
+            "maxLength": MAX_CHAIN_NAME_LENGTH,
```

Comment:
because if the `ownChainAccount` does not exist (can only happen on a sidechain), I can set this property to the default value of empty string. Having a minimum length was a bit awkward because then the empty string would not follow the schema

Reply:
makes sense